rs/migration_canister/src/external_interfaces/management.rs

```diff
@@ -252,3 +256,25 @@ pub async fn rename_canister(
         }
     }
 }
+
+// ========================================================================= //
+// `list_canister_snapshots`
+
+pub async fn assert_no_snapshots(canister_id: Principal) -> ProcessingResult<(), ValidationError> {
+    match list_canister_snapshots(&ListCanisterSnapshotsArgs { canister_id }).await {
+        Ok(snapshots) => {
+            if snapshots.is_empty() {
+                ProcessingResult::Success(())
+            } else {
+                ProcessingResult::FatalFailure(ValidationError::TargetHasSnapshots)
+            }
+        }
+        Err(e) => {
+            println!(
+                "Call `list_canister_snapshots` for {:?} failed: {:?}",
```

Comment:
```suggestion
                "Call `list_canister_snapshots` for {} failed: {:?}",
```
otherwise the canister ID is printed like this:
```
Principal { len: 10, bytes: [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] }
```

Reply:
163643d275